src/Mobs/Monster.cpp

```diff
@@ -118,87 +122,56 @@ void cMonster::SpawnOn(cClientHandle & a_Client)
+		m_Path = new cPath(GetWorld(), Vector3d(floor(position.x), floor(position.y), floor(position.z)), Vector3d(floor(Dest.x), floor(Dest.y), floor(Dest.z)), 20);
+		
+		
+		m_IsFollowingPath = false;
+	}
+	m_PathStatus = m_Path->Step();
+	if (m_PathStatus == ePathFinderStatus::PATH_NOT_FOUND)
 	{
-		// Too low/high, can't really do anything
+		// printf("No PATH *******************...\n");
 		FinishPathFinding();
-		return;
 	}
-
-	for (size_t i = 0; i < ARRAYCOUNT(gCrossCoords); i++)
+	if (m_PathStatus == ePathFinderStatus::CALCULATING)
 	{
-		if (IsCoordinateInTraversedList(Vector3i(gCrossCoords[i].x + PosX, PosY, gCrossCoords[i].z + PosZ)))
-		{
-			continue;
-		}
-
-		BLOCKTYPE BlockAtY = m_World->GetBlock(gCrossCoords[i].x + PosX, PosY, gCrossCoords[i].z + PosZ);
-		BLOCKTYPE BlockAtYP = m_World->GetBlock(gCrossCoords[i].x + PosX, PosY + 1, gCrossCoords[i].z + PosZ);
-		BLOCKTYPE BlockAtYPP = m_World->GetBlock(gCrossCoords[i].x + PosX, PosY + 2, gCrossCoords[i].z + PosZ);
-		int LowestY = FindFirstNonAirBlockPosition(gCrossCoords[i].x + PosX, gCrossCoords[i].z + PosZ);
-		BLOCKTYPE BlockAtLowestY = (LowestY >= cChunkDef::Height) ? E_BLOCK_AIR : m_World->GetBlock(gCrossCoords[i].x + PosX, LowestY, gCrossCoords[i].z + PosZ);
-
-		if (
-			(!cBlockInfo::IsSolid(BlockAtY)) &&
-			(!cBlockInfo::IsSolid(BlockAtYP)) &&
-			(!IsBlockLava(BlockAtLowestY)) &&
-			(BlockAtLowestY != E_BLOCK_CACTUS) &&
-			(PosY - LowestY < FALL_DAMAGE_HEIGHT)
-			)
-		{
-			m_PotentialCoordinates.push_back(Vector3d((gCrossCoords[i].x + PosX), PosY, gCrossCoords[i].z + PosZ));
-		}
-		else if (
-			(cBlockInfo::IsSolid(BlockAtY)) &&
-			(BlockAtY != E_BLOCK_CACTUS) &&
-			(!cBlockInfo::IsSolid(BlockAtYP)) &&
-			(!cBlockInfo::IsSolid(BlockAtYPP)) &&
-			(BlockAtY != E_BLOCK_FENCE) &&
-			(BlockAtY != E_BLOCK_FENCE_GATE)
-			)
-		{
-			m_PotentialCoordinates.push_back(Vector3d((gCrossCoords[i].x + PosX), PosY + 1, gCrossCoords[i].z + PosZ));
-		}
+		// printf("calculating...\n");
+		m_Destination=GetPosition();
 	}
-
-	if (!m_PotentialCoordinates.empty())
+	if (m_PathStatus == ePathFinderStatus::PATH_FOUND)
 	{
-		Vector3f ShortestCoords = m_PotentialCoordinates.front();
-		for (std::vector<Vector3d>::const_iterator itr = m_PotentialCoordinates.begin(); itr != m_PotentialCoordinates.end(); ++itr)
+		if (ReachedDestination() || (m_IsFollowingPath == false))
+		{
+			// printf("Getting next point...\n");
+			m_Destination = m_Path->GetNextPoint();
+			m_IsFollowingPath = true;
+			m_GiveUpCounter=40;  // Give up after 2 seconds if failed to reach m_Dest
+		}
+		/*else
 		{
-			Vector3f Distance = m_FinalDestination - ShortestCoords;
-			Vector3f Distance2 = m_FinalDestination - *itr;
-			if (Distance.SqrLength() > Distance2.SqrLength())
+			if (!ReachedDestination())
 			{
-				ShortestCoords = *itr;
+				printf("Did not reach destination\n");
 			}
+			if (!m_IsFollowingPath)
+			{
+				printf("Not following.\n");
+			}
+		}*/
```

Comment:
I'd like to keep it just for a bit more, might need it. I will remove all the printfs just before we merge.